src/cobra/io/mat.py

```diff
@@ -179,6 +353,214 @@ def create_mat_metabolite_id(model: Model) -> str:
             yield met.id
 
 
+def mat_parse_annotations(
+    target_list: List[Object], mat_struct: np.ndarray, _d_replace: str = D_MET
+) -> None:
+    """Process mat structure annotations in place.
+
+    Will process mat structured annotations and add them to a list of new entities
+    (metabolites, reactions, genes) in a format based on identifiers.org.
+
+    Parameters
+    ----------
+    target_list: list[cobra.Object]
+        A list of cobra objects, including metabolites, reactions or genes. The
+        annotations will be added to these lists.
+    mat_struct: np.ndarray
+        A darray that includes the data imported from matlab file.
+    _d_replace: str
+        A string that points to the dictionary of converstions between MATLAB and
+        providers. Default D_MET (for metabolite).
+    """
+    annotation_matlab = list(
+        set(mat_struct.dtype.names).intersection(D_REPLACE[_d_replace].keys())
+    )
+    providers = [D_REPLACE[_d_replace][x] for x in annotation_matlab]
+    annotations = dict.fromkeys(providers, None)
+    _pumbed_re = re.compile("PMID: ?(\\d+),?")
+    _ec_re = re.compile(r"([\d\-]+.[\d\-]+.[\d\-]+.[\d-]+)")
+    _chebi_re = re.compile(r"\D*(\d+),?")
+    for name, mat_key in zip(providers, annotation_matlab):
+        if mat_key == "rxnReferences":
+            # This only picks up PMID: style references. Sometimes there are other
+            # things like PMC or OMIM, but those are ignored for now,
+            annotations[name] = _cell_to_str_list(
+                mat_struct[mat_key][0, 0], None, _pumbed_re
+            )
+        elif mat_key == "rxnECNumbers":
+            # turn EC codes to a list
+            annotations[name] = _cell_to_str_list(
+                mat_struct[mat_key][0, 0], None, _ec_re
+            )
+        elif mat_key == "metCHEBIID":
+            # if there are more than one CHEBI code, turn them to a comma separated str
+            annotations[name] = _cell_to_str_list(
+                mat_struct[mat_key][0, 0], None, _chebi_re
+            )
+        else:
+            # If it something else, which may have commas, turn it into a list
+            annotations[name] = [
+                [y.strip() for y in _each_cell.split(", ")] if _each_cell else None
+                for _each_cell in _cell_to_str_list(mat_struct[mat_key][0, 0])
+            ]
+    for i, obj in enumerate(target_list):
+        obj.annotation = {
+            prov: annotations[prov][i] for prov in providers if annotations[prov][i]
+        }
+
+
+def mat_parse_notes(
```

Comment:
Clarifying.
There are situations where the rxnReferences in matlab has something like
'PMID:9028. Also when we read this book by Blah et al, 2001, published Random House' This can't be an identifier, so I'm not moving it into annotations, but notes. I don't want to throw it away, since it is info

The mat_to_notes function takes the notes from three fields (rxnReferences as above, rxnConfidenceScores, rxnNotes), throws away anything that can go into annotations (that is in identifiers), and keeps the rest.
It is in a dictionary, because Confidence Scores are still kept as a distinct entity, even if COBRA objects dont' have it. Otherwise, we will lose data on round trip back and forth to mat, and some people care about Confidence Scores. 
SBML reader still reads notes into a dictionary as well.
It could be a string, but then read-write matlab would loose information, including confidence scores which people care about.

Reply:
Oh god, yeah I understand now. I guess it makes sense to have those in the notes then. But I would leave them there. The issue with confidence scores saved outside of notes or MIRIAM annotations is that there is no way to interpret them. For instance, what do confidences of 100, 0, and -10 mean if they appear? Is a confidence of 3 in model A the same as in model B? So in the end, that information is hard to use outside a standardized system.